dex/msgjson/types.go

```diff
@@ -367,63 +367,80 @@ func (msg *Message) String() string {
+// 	Matches      []*Match `json:"matches"`
+// }
+
+// var _ Signable = (*MatchSet)(nil)
+
+// // Serialize serializes the Match data.
+// func (m *MatchSet) Serialize() []byte {
+// 	s := make([]byte, 0, len(m.Matches)*matchSize+8+8)
+// 	s = append(s, uint64Bytes(m.FeeRateBase)...)
+// 	s = append(s, uint64Bytes(m.FeeRateQuote)...)
+// 	for _, mi := range m.Matches {
+// 		s = append(s, mi.Serialize()...)
+// 	}
+// 	return s
+// }
+
 // Match is the params for a DEX-originating MatchRoute request.
 type Match struct {
 	Signature
-	OrderID    Bytes  `json:"orderid"`
-	MatchID    Bytes  `json:"matchid"`
-	Quantity   uint64 `json:"qty"`
-	Rate       uint64 `json:"rate"`
-	Address    string `json:"address"`
-	ServerTime uint64 `json:"tserver"`
+	OrderID      Bytes  `json:"orderid"`
+	MatchID      Bytes  `json:"matchid"`
+	Quantity     uint64 `json:"qty"`
+	Rate         uint64 `json:"rate"`
+	ServerTime   uint64 `json:"tserver"`
+	Address      string `json:"address"`
+	FeeRateBase  uint64 `json:"feeratebase"`
+	FeeRateQuote uint64 `json:"feeratequote"`
 	// Status and Side are provided for convenience and are not part of the
 	// match serialization.
 	Status uint8 `json:"status"`
 	Side   uint8 `json:"side"`
 }
 
-var _ Stampable = (*Match)(nil)
+var _ Signable = (*Match)(nil)
 
 // Serialize serializes the Match data.
 func (m *Match) Serialize() []byte {
 	// Match serialization is orderid (32) + matchid (32) + quantity (8) + rate (8)
-	// + server time (8) + address (variable, guess 35). Sum = 123
-	s := make([]byte, 0, 123)
+	// + server time (8) + address (variable, guess 35) + base fee rate (8) +
+	// quote fee rate (8) = 139
+	s := make([]byte, 0, 139)
 	s = append(s, m.OrderID...)
 	s = append(s, m.MatchID...)
 	s = append(s, uint64Bytes(m.Quantity)...)
 	s = append(s, uint64Bytes(m.Rate)...)
 	s = append(s, uint64Bytes(m.ServerTime)...)
-	return append(s, []byte(m.Address)...)
-}
-
-// Stamp sets the server timestamp and epoch ID. Partially satisfies the
-// Stampable interface.
-func (m *Match) Stamp(t uint64) {
-	m.ServerTime = t
+	s = append(s, []byte(m.Address)...)
+	s = append(s, uint64Bytes(m.FeeRateBase)...)
+	return append(s, uint64Bytes(m.FeeRateQuote)...)
 }
 
 // Init is the payload for a client-originating InitRoute request.
 type Init struct {
 	Signature
-	OrderID Bytes `json:"orderid"`
-	MatchID Bytes `json:"matchid"`
-	CoinID  Bytes `json:"coinid"`
-	// Time     uint64 `json:"timestamp"` // client originating
+	OrderID  Bytes `json:"orderid"`
+	MatchID  Bytes `json:"matchid"`
+	CoinID   Bytes `json:"coinid"`
 	Contract Bytes `json:"contract"`
 }
 
 var _ Signable = (*Init)(nil)
 
 // Serialize serializes the Init data.
 func (init *Init) Serialize() []byte {
-	// Init serialization is orderid (32) + matchid (32) + txid (probably 32) +
-	// vout (4) + contract (97 ish). Sum = 197
+	// Init serialization is orderid (32) + matchid (32) + coinid (probably 36)
+	// + contract (97 ish). Sum = 197
```

Comment:
`coinid (max 36) + contract (max 97)` would be more definitive assuming the sizes there are the maximum possible sizes.

Reply:
Unfortunately coinid could be 24 bytes or 240 bytes, just depends on how the asset represents it's funds.